lib/healthchecks/queue_latency_check.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module HealthChecks
+  class QueueLatencyCheck < OkComputer::Check
+    attr_accessor :threshold
+
+    def initialize(threshold = 30)
+      self.threshold = threshold.to_i
+    end
+
+    def check
+      @failures = nil
+      @message = Hash.new
+      queues = Sidekiq::Queue.all
+      queues.each do |q|
+        latency(q.name, threshold)
+      end
+
+      mark_failure if @failures
+      mark_message @message
+    end
```

Comment:
I think this can be simplified:
``` ruby 
def check
  @message = Hash.new
  Sidekiq::Queue.all.map do |queue|
    queue_latency = Sidekiq::Queue.new(queue).latency
    @message[queue] = "has a latency of #{queue_latency} seconds"
    mark_failure if queue_latency > threshold
  end
end
```
I don't believe `mark_message` is necessary because `message` has been declared an attr_accessor in the super class.

Reply:
Neat! I'm not sure why i didn't keep this as one method prior.